### PR TITLE
Sklearn update changes

### PR DESCRIPTION
**Pull request by neelasha23**
This PR accommodates for all changes made in latest sklearn version (1.2.0)
1. `n_init` param added to KMeans initialisation.
2. `normalise` parameter is no longer supported in `LinearRegression`, so removed.
3. `load_boston` dataset deprecated, hence replaced with `fetch_california_housing`

### CODE REVIEW

File: docs/source/nbs/NotebookCollection.ipynb [hunk 1]
```diff
@@ -70,9 +127,7 @@
     "    }\n",
     "}, {\n",
     "    'model': 'sklearn.linear_model.LinearRegression',\n",
```
Comment (idomic) [inline docs/source/nbs/NotebookCollection.ipynb:129]: Here you removed the parameter on purpose?

**Review by idomic — APPROVED** [review]
looks good besides this missing param?
